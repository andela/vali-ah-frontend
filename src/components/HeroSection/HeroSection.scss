@import '~scss/variables';

.hero-section {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;

  &__hero-content {
    width: 45%;

    &__text {
      font-size: 2.5rem;
      line-height: 4rem;
    }

    &__subtext {
      font-size: 1.5rem;
      font-weight: 100;
    }

    &__button-wrapper {
      margin: 2rem 0;

      &__cta-button {
        background: $primarycolor;
        width: 10rem;
        padding: 1rem;
        border: 0.0625rem solid $primarycolor;
        border-radius: 0.5rem;
        font-family: 'Rubik', sans-serif;
        font-size: 1rem;
        text-align: center;
        box-shadow: 0 0.625rem 2rem rgba(0, 0, 0, 0.22);
        cursor: pointer;
        color: $whitecolor;
        transition: 0.3s;
      }

      &--signin-button {
        background: none;
        margin-left: 2rem;
        border: 0.0625rem solid $primarycolor;
        border-radius: 0.5rem;
        box-shadow: none;
        color: $primarycolor;
      }

      &--signin-button:hover {
        background: $hovercolor;
        color: $whitecolor;
      }

      &__cta-button:hover {
        background: $hovercolor;
        box-shadow: 0 0.625rem 2rem rgba(0, 0, 0, 0.22);
        opacity: 1;
      }
    }
  }

  &__hero-image {
    width: 37.5rem;

    &__image-width {
      width: 100%;
    }
  }

  &__kicker-wrapper {
    display: flex;
    align-items: end;

    &__kicker-text {
      margin-right: 1rem;
      color: $primarycolor;
    }

    &__scrolldown {
      width: 4%;
    }
  }
}

@media (max-width: 48.0625rem) {
  .hero-section {
    flex-direction: column;
    padding: 1rem;

    &__hero-content {
      width: 100%;
      text-align: center;
  
      &__text {
        font-size: 1.2rem;
        line-height: 2rem;
      }
  
      &__subtext {
        font-size: 1rem;
      }
  
      &__button-wrapper {
        display: flex;
        justify-content: center;

        &__cta-button {
          font-size: 0.9rem;
        }

        &--signin-button {
          font-size: 0.9rem;
        }
      }
    }

    &__kicker-wrapper {
      display: block;
    }

    &__hero-image {
      width: 16rem;
    }
  }
}
