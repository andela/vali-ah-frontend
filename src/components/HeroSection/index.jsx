import React from 'react';

import './HeroSection.scss';

import heroImage from '../../assets/images/HeroImage.jpg';
import downArrowIcon from '../../assets/images/arrow-down.svg';

const HeroSection = () => (
  <section className="app-container hero-section">
    <div className="hero-section__hero-content">
      <h1 className="hero-section__hero-content__text"> The one-stop shop for all the self-help ideas you need </h1>
      <p className="hero-section__herocontent__subtext"> Write and share your stories </p>
      <div className="hero-section__hero-content__button-wrapper">
        <button type="button" className="hero-section__hero-content__button-wrapper__cta-button">Join 1kbIdeas</button>
        <button type="button" className="hero-section__hero-content__button-wrapper__cta-button hero-section__hero-content__button-wrapper--signin-button">Sign in</button>
      </div>
      <div className="hero-section__kicker-wrapper">
        <p className="hero-section__kicker-wrapper__kicker-text">Explore amazing topics</p>
        <img src={downArrowIcon} className="hero-section__kicker-wrapper__scrolldown" alt="Scrolldown Icon" />
      </div>
    </div>
    <div className="hero-section__hero-image">
      <img src={heroImage} alt="1kbIdeas community" className="hero-section__hero-image__image-width" />
    </div>
  </section>
);

export default HeroSection;
