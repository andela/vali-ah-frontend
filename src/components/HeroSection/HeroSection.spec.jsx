import React from 'react';

import HeroSection from './index';


describe('HeroSection Component', () => {
  it('should renders properly without card data', () => {
    const heroSection = mount(<HeroSection />);

    expect(heroSection).toMatchSnapshot();
  });
});
