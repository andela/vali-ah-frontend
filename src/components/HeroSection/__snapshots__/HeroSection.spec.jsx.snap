// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection Component should renders properly without card data 1`] = `
<HeroSection>
  <section
    className="app-container hero-section"
  >
    <div
      className="hero-section__hero-content"
    >
      <h1
        className="hero-section__hero-content__text"
      >
         The one-stop shop for all the self-help ideas you need 
      </h1>
      <p
        className="hero-section__herocontent__subtext"
      >
         Write and share your stories 
      </p>
      <div
        className="hero-section__hero-content__button-wrapper"
      >
        <button
          className="hero-section__hero-content__button-wrapper__cta-button"
          type="button"
        >
          Join 1kbIdeas
        </button>
        <button
          className="hero-section__hero-content__button-wrapper__cta-button hero-section__hero-content__button-wrapper--signin-button"
          type="button"
        >
          Sign in
        </button>
      </div>
      <div
        className="hero-section__kicker-wrapper"
      >
        <p
          className="hero-section__kicker-wrapper__kicker-text"
        >
          Explore amazing topics
        </p>
        <img
          alt="Scrolldown Icon"
          className="hero-section__kicker-wrapper__scrolldown"
          src="test-file-stub"
        />
      </div>
    </div>
    <div
      className="hero-section__hero-image"
    >
      <img
        alt="1kbIdeas community"
        className="hero-section__hero-image__image-width"
        src="test-file-stub"
      />
    </div>
  </section>
</HeroSection>
`;
